src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -151,18 +157,35 @@ class AppsResource @Inject() (
   @Timed
   def replaceMultiple(@DefaultValue("false")@QueryParam("force") force: Boolean,
                       body: Array[Byte],
-                      @Context req: HttpServletRequest, @Context resp: HttpServletResponse): Response = {
+                      @Context req: HttpServletRequest): Response = doIfAuthenticated(req) { implicit identity =>
+    import scala.concurrent.ExecutionContext.Implicits.global
 
+    val version = clock.now()
     withValid(Json.parse(body).as[Seq[AppUpdate]].map(_.withCanonizedIds())) { updates =>
-      doIfAuthorized(req, resp, UpdateAppOrGroup, updates.flatMap(_.id): _*) { identity =>
-        val version = clock.now()
-        def updateGroup(root: Group): Group = updates.foldLeft(root) { (group, update) =>
-          update.id match {
-            case Some(id) => group.updateApp(id, updateOrCreate(id, _, update, version), version)
-            case None     => group
+      val appliedUpdates = result(
```

Comment:
To make this code transactional, you need to move the code int updateGroup (throw an Exception if not authorized).